src/main/java/cz/centrum/haffner/SimpleTrainingDavid/AppServices/MetricsInfoService.java

```diff
@@ -1,15 +1,176 @@
 package cz.centrum.haffner.SimpleTrainingDavid.AppServices;
 
-import cz.centrum.haffner.SimpleTrainingDavid.DataTemplates.MetricsInfoSet;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.core.type.TypeReference;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
 
 // algorithm for returning of the proper MetricsInfo set
 public class MetricsInfoService {
```

Comment:
V tuto chvíli MetricsInfoServise obsahuje logiku pro načítání souboru, parsování a drží informace o counterech.
Při každém requestu se tato servisa vytváří znovu...
Rozpadni ji na menší kusy - viz komentáře níž.
Annotuj ji jako @Service a místo abys ji při každém requestu vytvářel, tak ji jen použij v kontrolleru pro to, aby Ti vrátila výsledný objekt.

Reply:
snad hotovo - jen doufám, že jsem Tě správně pochopil:  teď ta servisa nuluje výsledný dataset (pro response) a pomocné countery na parseru, vše to předává a nechává na parseru a jen od něj dostává výsledný dataset pro controller;
případně ten parser ještě nějak rozdělím (načítání s parsováním + solo logika pro počítání dat pro dataset)

Reply:
Ano, servisu jsi vytvořil správně. Jako komponenta zawirovaná ve springu teď bude k dispozici pro provolání z controlleru.

Reply:
předěláno viz jiné komentáře ...